### DISCUSSION TITLE
When planning trajectories in a collision world, can objects the robot interacts be taken into account?

### DISCUSSION THREAD

**Discussion by bestgwl** [NVlabs/curobo#104] [Software Q&A] [answered]
When planning trajectories in a collision world, can objects the robot interacts be taken into account?

I'm trying to use a robot to grasp objects. Curobo can meet my needs very well, reach the specified grasp pose, and grab the object. However, since the trajectory planned later only considers the robot itself, not the object. Therefore, objects can easily collide during motion.
(For example, if you go to pick up a water cup from a drawer, the robotic arm can use curobo to enter the drawer without collision, but during the process of taking out the drawer, the water cup can easily hit the drawer directly).

This problem seems difficult to solve. Do you have any relevant ideas or features? Grateful

**Comment by balakumar-s** [comment, marked as answer]
Check this note: https://curobo.org/source/advanced_examples/2b_block_stacking_example.html

And this example: https://github.com/NVlabs/curobo/blob/main/examples/isaac_sim/simple_stacking.py
